File: Practice/17/Python/17.py
MAX_SIZE = 36
MIN_SIZE = 0
SIZE = abs(MAX_SIZE)+abs(MIN_SIZE)+1
MAX_PLAYS = 12 #K
LAYOUT = 'Z101010101001010101101010101001010101'

COLOUR_STACK = MAX_PLAYS*['-']

class NumberElement:
	counted: int = 0
	lastCount: int = -1

def calculate(a, id, playCounter):
	a[id].counted += 1
	a[id].lastCount = playCounter
	COLOUR_STACK.pop(0)
	COLOUR_STACK.append(LAYOUT[id])

def printMostCommonNumbers(a):
	max = -1
	for id in range(MIN_SIZE, MAX_SIZE+1):
		if a[id].counted > max:
			max = a[id].counted
	for id in range(MIN_SIZE, MAX_SIZE+1):
		if a[id].counted == max:
			print(id, ' ', sep='', end='')
	print()

def printMissingNumbers(a, playCounter):
	K = min(MAX_PLAYS-1, playCounter)
	for id in range(MIN_SIZE, MAX_SIZE+1):
		if playCounter-a[id].lastCount > K:
			print(id, ' ', sep='', end='')
	print()

def printColourStats(COLOUR_STACK):
	RED = 0
	BLACK = 0
	for id in range(0, len(COLOUR_STACK)):
		if COLOUR_STACK[id] == '0': BLACK += 1
		if COLOUR_STACK[id] == '1': RED +=1
	print(RED, BLACK) 
		

def printResult(a, playCounter):
	printMostCommonNumbers(a)
	printMissingNumbers(a, playCounter)
	printColourStats(COLOUR_STACK)
	print()

a = []
for i in range(0, SIZE):
	a.append(NumberElement())
playCounter = 0
#Input
id = int(input())
while not (id == -1):
	calculate(a, id, playCounter)
	printResult(a, playCounter)
	playCounter += 1
	id = int(input())

File: Practice/17/С++/17.cpp
#include <iostream>
using namespace std;
int main()
{
    const int N = 37;
    int A[N];
    int red[]{ 1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36 };
    int black[]{ 2, 4, 6, 8, 10, 11, 13, 15, 17, 20, 22, 24, 26, 28, 29, 31, 33, 35 };
    int i, n, b, t, s;
    b = 0;
    t = 0;
    s = 0;
    for (i = 0; i < N; i++)
    {
        A[i] = 0;
    }
    while (1)
    {
        cin >> n;
        for (i = 0; i < 18; i++)
        {
            if (n == red[i])
                s++;
            if (n == black[i])
                t++;

        }
        A[n]++;
        if (n < 0)
            break;
        for (int i = 0; i < N; i++)
        {
            if (A[i] > b)
                b = A[i];
        }
        for (int i = 0; i < N; i++)
        {
            if (A[i] == b) cout << i << " ";
        }
        cout << endl;
        for (int i = 0; i < N; i++)
        {
            if (A[i] == 0) cout << i << " ";
        }
        cout << endl << s << " " << t << endl;
    }




    system("pause");
    return 0;
}